src/sentry/static/sentry/app/views/organizationIntegrations/integrationListDirectory.tsx

```diff
@@ -221,6 +266,33 @@ class OrganizationIntegrations extends AsyncComponent<
       .sort((a, b) => this.getInstallValue(b) - this.getInstallValue(a));
   }
 
+  async componentDidUpdate(_, prevState: State) {
+    if (this.state.list.length !== prevState.list.length) {
+      await this.createSearch();
+    }
+  }
+
+  async createSearch() {
+    const {list} = this.state;
+    this.setState({
+      fuzzy: await createFuzzySearch(list || [], {
```

Comment:
@NisanthanNanthakumar I don't think fuzzy belongs as a state field since it isn't directly tied to the view layer. Would be better as a property of the class object.